src/Generator.cpp

```diff
@@ -944,55 +896,48 @@ int generate_filter_main_inner(int argc, char **argv, std::ostream &cerr) {
+    auto target_strings = split_string(generator_args["target"].string_value, ",");
+    std::vector<Target> targets;
+    for (const auto &s : target_strings) {
+        targets.emplace_back(s);
+    }
+
+    // extensions won't vary across multitarget output
+    std::map<Output, OutputInfo> output_info = get_output_info(targets[0]);
 
+    std::set<Output> outputs;
     if (emit_flags.empty() || (emit_flags.size() == 1 && emit_flags[0].empty())) {
         // If omitted or empty, assume .a and .h and registration.cpp
-        emit_options.emit_static_library = emit_options.emit_h = emit_options.emit_registration = true;
+        outputs.insert(Output::c_header);
+        outputs.insert(Output::registration);
+        outputs.insert(Output::static_library);
     } else {
-        // If anything specified, only emit what is enumerated
-        for (const std::string &opt : emit_flags) {
-            if (opt == "assembly") {
-                emit_options.emit_assembly = true;
-            } else if (opt == "bitcode") {
-                emit_options.emit_bitcode = true;
-            } else if (opt == "stmt") {
-                emit_options.emit_stmt = true;
-            } else if (opt == "html") {
-                emit_options.emit_stmt_html = true;
-            } else if (opt == "cpp") {
-                emit_options.emit_cpp = true;
-            } else if (opt == "py.cpp") {
-                emit_options.emit_python_extension = true;
-            } else if (opt == "o") {
-                emit_options.emit_o = true;
-            } else if (opt == "h") {
-                emit_options.emit_h = true;
-            } else if (opt == "static_library") {
-                emit_options.emit_static_library = true;
-            } else if (opt == "cpp_stub") {
-                emit_options.emit_cpp_stub = true;
-            } else if (opt == "schedule") {
-                emit_options.emit_schedule = true;
-            } else if (opt == "pytorch_wrapper") {
-                emit_options.emit_pytorch_wrapper = true;
-            } else if (opt == "featurization") {
-                emit_options.emit_featurization = true;
-            } else if (opt == "registration") {
-                emit_options.emit_registration = true;
-            } else if (!opt.empty()) {
-                cerr << "Unrecognized emit option: " << opt
-                     << " not one of [assembly, bitcode, cpp, h, html, o, static_library, stmt, cpp_stub, registration, pytorch_wrapper], ignoring.\n";
-            }
+        // Build a reverse lookup table. Allow some legacy aliases on the command line,
+        // to allow legacy build systems to work more easily.
+        std::map<std::string, Output> output_name_to_enum = {
+            {"cpp", Output::c_source},
+            {"h", Output::c_header},
+            {"html", Output::stmt_html},
+            {"o", Output::object},
+            {"py.c", Output::python_extension},
```

Comment:
I wonder if we should use the extension map to autopopulate this. E.g. so that I can say "s" or "ll" instead of assembly,llvm_assembly. It's easier to remember the file extensions that I want than the enum names.

Reply:
Hmm... worth thinking about, but easy enough to add later if we want.